sql/core/src/main/scala/org/apache/spark/sql/execution/python.scala

```diff
@@ -220,7 +220,12 @@ object EvaluatePython {
       ArrayBasedMapData(keys, values)
 
     case (c, StructType(fields)) if c.getClass.isArray =>
-      new GenericInternalRow(c.asInstanceOf[Array[_]].zip(fields).map {
+      val array = c.asInstanceOf[Array[_]]
+      assert(
+        array.length == fields.length,
+        s"Row length ${array.length} and schema length ${fields.length} don't match"
+      )
```

Comment:
I am wondering if we should throw `IllegalStateException` because `AssertionError` can be disabled.

Also, regarding the error message, I wondering if something like `The given row does not have the same number of values required by the schema. ${fields.length} values are expected. But, ${array.length} values are provided` maybe a better one.